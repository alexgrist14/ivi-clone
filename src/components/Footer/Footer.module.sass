.footer
  .footer__btns
    display: flex
    flex-direction: column
    gap: 12px
    
