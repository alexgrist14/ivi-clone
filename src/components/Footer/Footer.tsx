import { FC, useState } from "react";
import FooterTop from "./FooterTop/FooterTop";
import FooterBottom from "./FooterBottom/FooterBottom";
import FooterCopyright from "./FooterCopyright/FooterCopyright";
import SearchModal from "../SearchModal/SearchModal";
import CustomButton from "/src/UI/CustomButton/CustomButton";

const Footer: FC = () => {
  const [isSearchActive, setIsSearchActive] = useState(false);

  return (
    <>
      <footer className="container">
        <FooterTop />
        <FooterBottom />
        <FooterCopyright />
        <CustomButton
          clickCallback={() => {
            setIsSearchActive(true);
          }}
        >
          Search
        </CustomButton>
      </footer>
      {isSearchActive && (
        <SearchModal
          closeCallback={() => {
            setIsSearchActive(false);
          }}
        />
      )}
    </>
  );
};

export default Footer;
