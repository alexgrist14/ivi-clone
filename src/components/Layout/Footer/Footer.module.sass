.footer
  padding-block: 24px
