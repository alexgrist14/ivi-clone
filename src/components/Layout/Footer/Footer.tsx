import { FC } from "react";
import FooterTop from "./FooterTop/FooterTop";
import FooterBottom from "./FooterBottom/FooterBottom";
import FooterCopyright from "./FooterCopyright/FooterCopyright";
import { useAppSelector } from "/src/hooks/redux";
import styles from "./Footer.module.sass";
import Container from "/src/UI/Container/Container";

const Footer: FC = () => {
  const windowSize = useAppSelector((state) => state.windowSize);

  return (
    <footer className={styles.footer}>
      <Container>
        {windowSize.width > 1159 && <FooterTop />}
        {windowSize.width > 1159 && <FooterBottom />}
        <FooterCopyright />
      </Container>
    </footer>
  );
};

export default Footer;
