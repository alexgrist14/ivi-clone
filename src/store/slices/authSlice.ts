import {createSlice} from "@reduxjs/toolkit";
import type {PayloadAction} from "@reduxjs/toolkit";

const initialState: UserAuth ={
  isLogged: false,
  userEmail: "",
}

export interface UserAuth{
  isLogged: boolean;
  userEmail: string;
  role?: string;
}

export const authSlice = createSlice({
  name: "auth",
  initialState,
  reducers: {
    setAuth: (state, action: PayloadAction<UserAuth>) => {
      state.isLogged = action.payload.isLogged;
      state.userEmail = action.payload.userEmail;
      state.role = action.payload.role;
    },
    setRole:(state,action:PayloadAction<string | undefined>)=>{
      state.role = action.payload;
    }
  },
});

export const {setAuth,setRole} = authSlice.actions;
export default authSlice.reducer;
