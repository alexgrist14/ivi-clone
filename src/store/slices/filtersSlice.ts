import { createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import { IMovie } from "/src/types/IMovie";
import { IFilters } from "/src/types/IFilter";
import getSortedList from "/src/utils/filters/getSortedList";
import { defaultFilters } from "/src/utils/filters/filtersVariables";

export interface FiltersState {
  filters: IFilters;
  filteredMovies: IMovie[];
  sortingMethod: string;
  isMoviesLoading: boolean;
  moviesPage: number;
}

const initialState: FiltersState = {
  filters: defaultFilters,
  filteredMovies: [],
  sortingMethod: "assessments",
  isMoviesLoading: false,
  moviesPage: 1,
};

export const filtersSlice = createSlice({
  name: "locale",
  initialState,
  reducers: {
    setFilters: (state, action: PayloadAction<IFilters>) => {
      state.filters = action.payload;
    },
    setFilteredMovies: (state, action: PayloadAction<IMovie[]>) => {
      state.filteredMovies = getSortedList(state.sortingMethod, action.payload);
    },
    setSortingMethod: (state, action: PayloadAction<string>) => {
      state.sortingMethod = action.payload;
    },
    setIsMoviesLoading: (state, action: PayloadAction<boolean>) => {
      state.isMoviesLoading = action.payload;
    },
    setMoviesPage: (state, action: PayloadAction<number>) => {
      state.moviesPage = action.payload;
    },
  },
});

export const {
  setFilters,
  setFilteredMovies,
  setSortingMethod,
  setIsMoviesLoading,
  setMoviesPage,
} = filtersSlice.actions;
export default filtersSlice.reducer;
