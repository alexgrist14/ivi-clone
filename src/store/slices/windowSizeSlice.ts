import { createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import IWindowSize from "/src/types/IWindowSize";

const initialState: IWindowSize = { width: 0, height: 0 };

export const windowSizeSlice = createSlice({
  name: "windowSize",
  initialState,
  reducers: {
    setWindowSize: (state, action: PayloadAction<IWindowSize>) => {
      state.width = action.payload.width;
      state.height = action.payload.height;
    },
  },
});

export const { setWindowSize } = windowSizeSlice.actions;
export default windowSizeSlice.reducer;
