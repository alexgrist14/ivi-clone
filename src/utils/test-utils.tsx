import { ReactNode } from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { reducersSetup } from "/src/store";
import { configureStore } from "@reduxjs/toolkit";
import { IProviderRender } from "../types/IRender";

export const renderWithProviders = (children: ReactNode): IProviderRender => {
  const store = configureStore({ reducer: reducersSetup });
  const component = render(<Provider store={store}>{children}</Provider>);

  return { store, component };
};
