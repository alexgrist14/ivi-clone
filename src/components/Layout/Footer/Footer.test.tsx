import Footer from "./Footer";
import { renderWithProviders } from "/src/utils/test-utils";
import { act } from "@testing-library/react";
import { setWindowSize } from "/src/store/slices/windowSizeSlice";

describe("Footer widget", () => {
  test("should match to snapshot with width more than 1159", () => {
    const render = renderWithProviders(<Footer />);
    act(() =>
      render.store.dispatch(setWindowSize({ width: 1920, height: 1080 }))
    );
    expect(render.component.container).toMatchSnapshot();
  });

  test("should match to snapshot with width less than 1160px", () => {
    const render = renderWithProviders(<Footer />);
    act(() =>
      render.store.dispatch(setWindowSize({ width: 1159, height: 1080 }))
    );
    expect(render.component.container).toMatchSnapshot();
  });
});
