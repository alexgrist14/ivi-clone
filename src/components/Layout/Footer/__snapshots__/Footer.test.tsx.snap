// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer widget should match to snapshot with width less than 1160px 1`] = `
<div>
  <footer
    class="footer"
  >
    <div
      class="container"
      data-testid="container"
    >
      <aside
        class="copyright"
        data-testid="copyright"
      >
        <span>
          HBO Â® and related service marks are the property of Home Box Office, Inc
        </span>
      </aside>
    </div>
  </footer>
</div>
`;

exports[`Footer widget should match to snapshot with width more than 1159 1`] = `
<div>
  <footer
    class="footer"
  >
    <div
      class="container"
      data-testid="container"
    >
      <div
        class="top"
      >
        <div>
          <span
            class="top__title"
          >
            about.title
          </span>
          <nav
            class="top__nav"
          >
            <ul
              class="list"
              data-testid="footer-list"
            >
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://corp.ivi.ru/?_gl=1*15hvcv0*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTI5OTQ0Ny42MC4wLjA."
                  target="_blank"
                >
                  about.links.0
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://corp.ivi.ru/career/?_gl=1*2qtg9j*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTMwMTQwMy41Ny4wLjA.#career-vacancy-block"
                  target="_blank"
                >
                  about.links.1
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/pages/beta"
                  target="_blank"
                >
                  about.links.2
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/info/partners"
                  target="_blank"
                >
                  about.links.3
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://corp.ivi.ru/advertisers/?_gl=1*uhgz7x*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTMwMTQwMy41Ny4wLjA."
                  target="_blank"
                >
                  about.links.4
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/info/agreement"
                  target="_blank"
                >
                  about.links.5
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/info/confidential"
                  target="_blank"
                >
                  about.links.6
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/info/goryachaya-liniya-komplaens"
                  target="_blank"
                >
                  about.links.7
                </a>
              </li>
            </ul>
          </nav>
        </div>
        <div>
          <span
            class="top__title"
          >
            sections.title
          </span>
          <nav
            class="top__nav"
          >
            <ul
              class="list"
              data-testid="footer-list"
            >
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/"
                  target="_blank"
                >
                  sections.links.0
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/new"
                  target="_blank"
                >
                  sections.links.1
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="/movies"
                >
                  sections.links.2
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/series"
                  target="_blank"
                >
                  sections.links.3
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/animation"
                  target="_blank"
                >
                  sections.links.4
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/tvplus"
                  target="_blank"
                >
                  sections.links.5
                </a>
              </li>
              <li
                class="list__item"
              >
                <a
                  class="link"
                  href="https://www.ivi.ru/goodmovies"
                  target="_blank"
                >
                  sections.links.6
                </a>
              </li>
            </ul>
            <div
              class="cert"
            >
              <a
                class="cert__link"
                href="https://www.ivi.ru/cert"
                target="_blank"
              >
                cert
              </a>
            </div>
          </nav>
        </div>
        <div>
          <span
            class="top__title"
          >
            support.title
          </span>
          <div
            class="top__wrapper"
          >
            <p>
              support.text
            </p>
            <div
              class="btns"
              data-testid="footer-buttons"
            >
              <button
                class="btn  unselectable"
                data-testid="chat"
                style="width: 196px;"
              >
                <div
                  class="btn__content"
                >
                  chat
                </div>
              </button>
              <div
                class="btns__wrapper"
              >
                <button
                  class=" btn btn_icon  unselectable"
                  data-testid="support"
                >
                  <div
                    class="btn__content"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      height="20"
                      stroke="currentColor"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                  </div>
                </button>
                <div>
                  <button
                    class=" btn btn_icon  unselectable"
                    data-testid="phones"
                  >
                    <div
                      class="btn__content"
                    >
                      <svg
                        fill="currentColor"
                        height="16"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3.654 1.328a.678.678 0 0 0-1.015-.063L1.605 2.3c-.483.484-.661 1.169-.45 1.77a17.568 17.568 0 0 0 4.168 6.608 17.569 17.569 0 0 0 6.608 4.168c.601.211 1.286.033 1.77-.45l1.034-1.034a.678.678 0 0 0-.063-1.015l-2.307-1.794a.678.678 0 0 0-.58-.122l-2.19.547a1.745 1.745 0 0 1-1.657-.459L5.482 8.062a1.745 1.745 0 0 1-.46-1.657l.548-2.19a.678.678 0 0 0-.122-.58L3.654 1.328zM1.884.511a1.745 1.745 0 0 1 2.612.163L6.29 2.98c.329.423.445.974.315 1.494l-.547 2.19a.678.678 0 0 0 .178.643l2.457 2.457a.678.678 0 0 0 .644.178l2.189-.547a1.745 1.745 0 0 1 1.494.315l2.306 1.794c.829.645.905 1.87.163 2.611l-1.034 1.034c-.74.74-1.846 1.065-2.877.702a18.634 18.634 0 0 1-7.01-4.42 18.634 18.634 0 0 1-4.42-7.009c-.362-1.03-.037-2.137.703-2.877L1.885.511z"
                        />
                      </svg>
                    </div>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="ask"
              data-testid="footer-ask"
            >
              <a
                class="ask__link"
                href="https://ask.ivi.ru/?_gl=1*17d3i43*_ga*MTYyOTgwODIxOC4xNjgxMjEzNzAx*_ga_GETQ4387MJ*MTY4MTI5Mzg5My4zLjEuMTY4MTMwNDIxNS41NS4wLjA."
                target="_blank"
              >
                ask.ivi.ru
              </a>
              <p
                class="ask__text"
              >
                ask
              </p>
            </div>
          </div>
        </div>
        <div>
          <a
            class="widget"
            href="https://www.ivi.ru/subscribe?redirect_url=%2F"
            target="_blank"
          >
            <div
              class="widget__icon"
            >
              <div
                class="widget__svg"
              >
                <svg
                  fill="currentColor"
                  height="56"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="56"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    id="Bullhorn"
                  >
                    <path
                      d="M4.5,16.474c-0.849,-0.4 -1.436,-1.263 -1.436,-2.264l-0,-4.419c-0,-1.38 1.118,-2.5 2.5,-2.5l4.343,0c0.793,0 1.581,-0.132 2.33,-0.392c1.859,-0.705 3.792,-1.727 5.24,-2.922l0.869,-0.718c0.015,-0.013 0.032,-0.025 0.049,-0.036c0.666,-0.424 1.538,0.054 1.538,0.844l0,6.717c0.572,0.11 1.004,0.613 1.004,1.217c-0,0.604 -0.432,1.106 -1.004,1.216l-0,6.718c-0,0.787 -0.872,1.267 -1.538,0.843c-0.017,-0.011 -0.034,-0.023 -0.05,-0.036l-0.868,-0.718c-1.446,-1.195 -3.364,-2.214 -5.226,-2.891c-0.748,-0.261 -1.536,-0.394 -2.328,-0.394c-0.609,-0.029 -1.265,-0.029 -1.265,-0.029l0,2.147c0,1.148 -0.931,2.079 -2.079,2.079c-1.148,-0 -2.079,-0.931 -2.079,-2.079l0,-2.383Zm1,0.236l0,2.147c0,0.596 0.483,1.079 1.079,1.079c0.596,-0 1.079,-0.483 1.079,-1.079c0,-0 0,-2.147 0,-2.147l-2.094,-0c-0.031,-0 -0.053,-0 -0.064,-0Zm6,-0.882l0.142,0.04c2.37,0.664 4.575,1.817 6.473,3.385l0.818,0.677l-0,-15.859l-0.82,0.677c-1.897,1.566 -4.1,2.717 -6.468,3.379l-0.145,0.041l-0,7.66Zm-2.842,-0.118l1.842,0l-0,-7.419l-4.936,0c-0.829,0 -1.5,0.672 -1.5,1.5l-0,4.419c-0,0.829 0.671,1.5 1.499,1.5l3.095,0Z"
                    />
                  </g>
                </svg>
                <svg
                  class="widget__slash"
                  fill="currentColor"
                  height="48"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 16 16"
                  width="48"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13.854 2.146a.5.5 0 0 1 0 .708l-11 11a.5.5 0 0 1-.708-.708l11-11a.5.5 0 0 1 .708 0Z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
            <p
              class="widget__text"
            >
              widget
            </p>
          </a>
        </div>
      </div>
      <div
        class="bottom"
        data-testid="footer-bottom"
      >
        <div
          class="bottom__left"
        >
          <button
            class="btn  unselectable"
          >
            <div
              class="btn__content"
            >
              <div
                class="btn"
              >
                <div
                  class="btn__icon"
                >
                  <svg
                    fill="currentColor"
                    height="20"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 16 16"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.182.008C11.148-.03 9.923.023 8.857 1.18c-1.066 1.156-.902 2.482-.878 2.516.024.034 1.52.087 2.475-1.258.955-1.345.762-2.391.728-2.43Zm3.314 11.733c-.048-.096-2.325-1.234-2.113-3.422.212-2.189 1.675-2.789 1.698-2.854.023-.065-.597-.79-1.254-1.157a3.692 3.692 0 0 0-1.563-.434c-.108-.003-.483-.095-1.254.116-.508.139-1.653.589-1.968.607-.316.018-1.256-.522-2.267-.665-.647-.125-1.333.131-1.824.328-.49.196-1.422.754-2.074 2.237-.652 1.482-.311 3.83-.067 4.56.244.729.625 1.924 1.273 2.796.576.984 1.34 1.667 1.659 1.899.319.232 1.219.386 1.843.067.502-.308 1.408-.485 1.766-.472.357.013 1.061.154 1.782.539.571.197 1.111.115 1.652-.105.541-.221 1.324-1.059 2.238-2.758.347-.79.505-1.217.473-1.282Z"
                    />
                    <path
                      d="M11.182.008C11.148-.03 9.923.023 8.857 1.18c-1.066 1.156-.902 2.482-.878 2.516.024.034 1.52.087 2.475-1.258.955-1.345.762-2.391.728-2.43Zm3.314 11.733c-.048-.096-2.325-1.234-2.113-3.422.212-2.189 1.675-2.789 1.698-2.854.023-.065-.597-.79-1.254-1.157a3.692 3.692 0 0 0-1.563-.434c-.108-.003-.483-.095-1.254.116-.508.139-1.653.589-1.968.607-.316.018-1.256-.522-2.267-.665-.647-.125-1.333.131-1.824.328-.49.196-1.422.754-2.074 2.237-.652 1.482-.311 3.83-.067 4.56.244.729.625 1.924 1.273 2.796.576.984 1.34 1.667 1.659 1.899.319.232 1.219.386 1.843.067.502-.308 1.408-.485 1.766-.472.357.013 1.061.154 1.782.539.571.197 1.111.115 1.652-.105.541-.221 1.324-1.059 2.238-2.758.347-.79.505-1.217.473-1.282Z"
                    />
                  </svg>
                </div>
                <div
                  class="btn__text"
                >
                  <span
                    class="btn__top"
                  >
                    appstore
                  </span>
                  <span
                    class="btn__bottom"
                  >
                    App Store
                  </span>
                </div>
              </div>
            </div>
          </button>
          <button
            class="btn  unselectable"
          >
            <div
              class="btn__content"
            >
              <div
                class="btn"
              >
                <div
                  class="btn__icon"
                >
                  <svg
                    fill="currentColor"
                    height="18"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M325.3 234.3L104.6 13l280.8 161.2-60.1 60.1zM47 0C34 6.8 25.3 19.2 25.3 35.3v441.3c0 16.1 8.7 28.5 21.7 35.3l256.6-256L47 0zm425.2 225.6l-58.9-34.1-65.7 64.5 65.7 64.5 60.1-34.1c18-14.3 18-46.5-1.2-60.8zM104.6 499l280.8-161.2-60.1-60.1L104.6 499z"
                    />
                  </svg>
                </div>
                <div
                  class="btn__text"
                >
                  <span
                    class="btn__top"
                  >
                    google
                  </span>
                  <span
                    class="btn__bottom"
                  >
                    Google Play
                  </span>
                </div>
              </div>
            </div>
          </button>
          <button
            class="btn  unselectable"
          >
            <div
              class="btn__content"
            >
              <div
                class="btn"
              >
                <div
                  class="btn__icon"
                >
                  <svg
                    fill="currentColor"
                    height="20"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M488 384H24a8 8 0 01-8-8V88a8 8 0 018-8h464a8 8 0 018 8v288a8 8 0 01-8 8z"
                    />
                    <rect
                      height="32"
                      rx="4"
                      ry="4"
                      width="288"
                      x="112"
                      y="400"
                    />
                  </svg>
                </div>
                <div
                  class="btn__text"
                >
                  <span
                    class="btn__top"
                  >
                    smarttv
                  </span>
                  <span
                    class="btn__bottom"
                  >
                    Smart TV
                  </span>
                </div>
              </div>
            </div>
          </button>
          <button
            class="btn  unselectable"
          >
            <div
              class="btn__content"
            >
              <div
                class="btn"
              >
                <div
                  class="btn__icon"
                >
                  <svg
                    fill="currentColor"
                    height="20"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M4 6h18V4H4c-1.1 0-2 .9-2 2v11H0v3h14v-3H4V6zm19 2h-6c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h6c.55 0 1-.45 1-1V9c0-.55-.45-1-1-1zm-1 9h-4v-7h4v7z"
                    />
                  </svg>
                </div>
                <div
                  class="btn__text"
                >
                  <span
                    class="btn__bottom"
                  >
                    devices
                  </span>
                </div>
              </div>
            </div>
          </button>
        </div>
        <div
          class="bottom__right"
        >
          <button
            class=" btn btn_icon  unselectable"
            data-testid="link-0"
            style="border-radius: 50%;"
          >
            <div
              class="btn__content"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M545 117.7c3.7-12.5 0-21.7-17.8-21.7h-58.9c-15 0-21.9 7.9-25.6 16.7 0 0-30 73.1-72.4 120.5-13.7 13.7-20 18.1-27.5 18.1-3.7 0-9.4-4.4-9.4-16.9V117.7c0-15-4.2-21.7-16.6-21.7h-92.6c-9.4 0-15 7-15 13.5 0 14.2 21.2 17.5 23.4 57.5v86.8c0 19-3.4 22.5-10.9 22.5-20 0-68.6-73.4-97.4-157.4-5.8-16.3-11.5-22.9-26.6-22.9H38.8c-16.8 0-20.2 7.9-20.2 16.7 0 15.6 20 93.1 93.1 195.5C160.4 378.1 229 416 291.4 416c37.5 0 42.1-8.4 42.1-22.9 0-66.8-3.4-73.1 15.4-73.1 8.7 0 23.7 4.4 58.7 38.1 40 40 46.6 57.9 69 57.9h58.9c16.8 0 25.3-8.4 20.4-25-11.2-34.9-86.9-106.7-90.3-111.5-8.7-11.2-6.2-16.2 0-26.2.1-.1 72-101.3 79.4-135.6z"
                />
              </svg>
            </div>
          </button>
          <button
            class=" btn btn_icon  unselectable"
            data-testid="link-1"
            style="border-radius: 50%;"
          >
            <div
              class="btn__content"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M275.1 334c-27.4 17.4-65.1 24.3-90 26.9l20.9 20.6 76.3 76.3c27.9 28.6-17.5 73.3-45.7 45.7-19.1-19.4-47.1-47.4-76.3-76.6L84 503.4c-28.2 27.5-73.6-17.6-45.4-45.7 19.4-19.4 47.1-47.4 76.3-76.3l20.6-20.6c-24.6-2.6-62.9-9.1-90.6-26.9-32.6-21-46.9-33.3-34.3-59 7.4-14.6 27.7-26.9 54.6-5.7 0 0 36.3 28.9 94.9 28.9s94.9-28.9 94.9-28.9c26.9-21.1 47.1-8.9 54.6 5.7 12.4 25.7-1.9 38-34.5 59.1zM30.3 129.7C30.3 58 88.6 0 160 0s129.7 58 129.7 129.7c0 71.4-58.3 129.4-129.7 129.4s-129.7-58-129.7-129.4zm66 0c0 35.1 28.6 63.7 63.7 63.7s63.7-28.6 63.7-63.7c0-35.4-28.6-64-63.7-64s-63.7 28.6-63.7 64z"
                />
              </svg>
            </div>
          </button>
          <button
            class=" btn btn_icon  unselectable"
            data-testid="link-2"
            style="border-radius: 50%;"
          >
            <div
              class="btn__content"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                />
              </svg>
            </div>
          </button>
          <button
            class=" btn btn_icon  unselectable"
            data-testid="link-3"
            style="border-radius: 50%;"
          >
            <div
              class="btn__content"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M20 15.5c-1.25 0-2.45-.2-3.57-.57a1.02 1.02 0 00-1.02.24l-2.2 2.2a15.045 15.045 0 01-6.59-6.59l2.2-2.21a.96.96 0 00.25-1A11.36 11.36 0 018.5 4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1 0 9.39 7.61 17 17 17 .55 0 1-.45 1-1v-3.5c0-.55-.45-1-1-1zM19 12h2a9 9 0 00-9-9v2c3.87 0 7 3.13 7 7zm-4 0h2c0-2.76-2.24-5-5-5v2c1.66 0 3 1.34 3 3z"
                />
              </svg>
            </div>
          </button>
          <button
            class=" btn btn_icon  unselectable"
            data-testid="link-4"
            style="border-radius: 50%;"
          >
            <div
              class="btn__content"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                />
              </svg>
            </div>
          </button>
          <button
            class=" btn btn_icon  unselectable"
            data-testid="link-5"
            style="border-radius: 50%;"
          >
            <div
              class="btn__content"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="padding-right: 3px;"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M446.7 98.6l-67.6 318.8c-5.1 22.5-18.4 28.1-37.3 17.5l-103-75.9-49.7 47.8c-5.5 5.5-10.1 10.1-20.7 10.1l7.4-104.9 190.9-172.5c8.3-7.4-1.8-11.5-12.9-4.1L117.8 284 16.2 252.2c-22.1-6.9-22.5-22.1 4.6-32.7L418.2 66.4c18.4-6.9 34.5 4.1 28.5 32.2z"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
      <aside
        class="copyright"
        data-testid="copyright"
      >
        <span>
          <span>
            Â© 
            2023
             ÐÐÐ Â«
            copyright
            Â»
            <br />
          </span>
          HBO Â® and related service marks are the property of Home Box Office, Inc
        </span>
      </aside>
    </div>
  </footer>
</div>
`;
